### PR TITLE
godot_4: init at 4.0-beta3

### PR DESCRIPTION
**Pull request by shiryel**
###### Description of changes

Godot 4, with the same [commit that beta 3 is using](https://godotengine.org/article/dev-snapshot-godot-4-0-beta-3).

I also moved Godot 3 into its own folder to keep both versions organized.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by rapenne-s** [comment]
Result of `nixpkgs-review pr 196881` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>godot_4</li>
  </ul>
</details>

File: pkgs/development/tools/godot_4/default.nix [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    cp bin/godot.* $out/bin/godot
+
+    #
+    # Install manual
+    #
+    installManPage misc/dist/linux/godot.6
+
+    #
+    # Install shortcut
+    #
```
Comment (SuperSandro2000) [inline pkgs/development/tools/godot_4/default.nix:118]:
```suggestion
    mkdir -p "$out/bin"
    cp bin/godot.* $out/bin/godot
    installManPage misc/dist/linux/godot.6
```

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -7269,13 +7269,15 @@ with pkgs;
 
   gocryptfs = callPackage ../tools/filesystems/gocryptfs { };
 
-  godot = callPackage ../development/tools/godot {};
+  godot_4 = callPackage ../development/tools/godot_4 { };
 
-  godot-export-templates = callPackage ../development/tools/godot/export-templates.nix { };
+  godot = callPackage ../development/tools/godot_3 { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:7274]:
```suggestion
  godot = callPackage ../development/tools/godot/3 { };
```
How about we move this into a subdirectory or use the following:

```suggestion
  godot = callPackage ../development/tools/godot/3.nix { };
```

Comment (shiryel) [reply pkgs/top-level/all-packages.nix:7274]: I think the subdirectory way is better, as the godot 4 variants will differ from godot 3. Also, thanks for the review.

Comment (ners) [reply pkgs/top-level/all-packages.nix:7274]: Consider making the directories `godot/godot_3` and `godot/godot_4`.

**Comment by tengkuizdihar** [comment]
`nixpkgs-review pr 196881` building as expected, running as expected.

**Pull request merged**